pkg/hub/addon/discovery_controller.go

```diff
@@ -110,48 +112,68 @@ func (c *addOnFeatureDiscoveryController) sync(ctx context.Context, syncCtx fact
 func (c *addOnFeatureDiscoveryController) syncAddOn(ctx context.Context, clusterName, addOnName string) error {
 	klog.V(4).Infof("Reconciling addOn %q", addOnName)
 
-	labels := map[string]string{}
```

Comment:
I do not remember why we need syncAddOn, seems to me syncCluster is sufficient...

Reply:
From integration testing, the `syncAddOn` is covered, so it is still being used?